src/euler.jl

```diff
@@ -0,0 +1,97 @@
+# This file is a part of AstroLib.jl. License is MIT "Expat".
+
+function _euler{T<:AbstractFloat}(ai::T, bi::T, select::Integer, FK4::Bool, radian::Bool)
+
+    if select>6 || select<1
+        error("Input for coordinate transformation should be an integer in the range 1:6")
+    end
+
+    if FK4
+        psi = [0.57595865315, 4.9261918136, 0, 0, 0.11129056012, 4.7005372834]
+        stheta = [0.88781538514, -0.88781538514,0.39788119938, -0.39788119938,
+                  0.86766174755, -0.86766174755]
+        ctheta = [0.46019978478, 0.46019978478, 0.91743694670, 0.9174369467,
+                  0.49715499774, 0.49715499774]
+        phi = [4.9261918136,  0.57595865315, 0, 0, 4.7005372834, 0.11129056012]
+    else
+        psi = [0.574770433, 4.9368292465, 0, 0, 0.11142137093, 4.71279419371]
+        stheta = [0.88998808748, -0.88998808748, 0.39777715593, -0.39777715593,
+                  0.86766622025, -0.86766622025]
+        ctheta = [0.45598377618, 0.45598377618, 0.91748206207, 0.91748206207,
+                  0.49714719172, 0.49714719172]
+        phi = [4.9368292465, 0.574770433, 0, 0, 4.71279419371, 0.11142137093]
+    end
+
+    if radian
+        ao = ai - phi[select]
+        cb = cos(bi)
+        x = (cb*sin(ao), cb*cos(ao), sin(bi))
+    else
+        ao = deg2rad(ai) - phi[select]
+        cb = cosd(bi)
+        x = (cb*sin(ao), cb*cos(ao), sind(bi))
+    end
+    bo = ctheta[select]*x[3] - stheta[select]*x[1]
+    ao = mod2pi(atan2(ctheta[select]*x[1] + stheta[select]*x[3], x[2]) + psi[select])
+    bo = asin(bo)
+
+    if radian
+        return (ao, bo)
+    end
+    return (rad2deg(ao), rad2deg(bo))
+end
+
+"""
+    euler(ai, bi, select[, FK4=true, radian=true])
+
+### Purpose ###
+
+Transform between Galactic, celestial, and ecliptic coordinates.
+
+### Explanation ###
+
+The function is used by the [astro](@ref) procedure.
+
+### Arguments ###
+
+* `ai`: input longitude, scalar or vector.
```

Comment:
Added. Small note -
I didn't use the new type system capabilities because of this line `typeai = typeof(float(one(R)))`. Also, modified AbstractArray to AbstractVector for more precise behavior.
Using the new method instead of dot call gives huge boost in speed (by @time)